azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -330,18 +330,22 @@ def _setup_azure_slice():
         def _reset_agent_cgroup_setup(self):
             try:
                 agent_drop_in_path = systemd.get_agent_drop_in_path()
-                if os.path.exists(agent_drop_in_path) and os.path.isdir(agent_drop_in_path):
+                if os.path.exists(agent_drop_in_path) and os.path.isdir(agent_drop_in_path) and len(os.listdir(agent_drop_in_path)) > 0:
```

Comment:
we shouldn't attempt agent cgroup setup cleanup if no drop-files exist in the folder when distro goes from supported to unsupported list